### PR TITLE
Migrate to new Kotlin 1.4.0-rc version and simplify maven publishing.

### PR DESCRIPTION
**Pull request by terrakok**

### CODE REVIEW

**Comment by terrakok** [comment]
It is proof of concept for simplify migration to Kotlin 1.4 in near future :)

**Comment by terrakok** [comment]
Update to Kotlin 1.4.0-rc.
I think you can publish rc version of Napier for testing.
New Kotlin version is coming :)

File: gradle/publish.gradle [hunk 1]
```diff
@@ -60,55 +64,10 @@ bintray {
...
-                def root = asNode()
-                root.appendNode('name', projectName)
-                root.appendNode('description', POM_DESCRIPTION)
-                root.appendNode('url', SITE_URL)
-                root.children().last() + pomConfig
-            }
-
-            // rename artifacts
-            groupId = BINTRAY_PACKAGE
-            if (name == 'kotlinMultiplatform') {
```
Comment (AAkira) [inline gradle/publish.gradle:94]: I think we still need these lines.
I know we write dependencies only one time in a common module on Kotlin 1.4.

Kotlin Coroutines has module files in core artifact.
https://repo1.maven.org/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-core/1.3.9/
But I can't make this file on Kotlin 1.4.
Do you have any idea?

**Comment by AAkira** [comment]
Thank you for preparing this.
I changed the target branch.

File: gradle/publish.gradle (hunk 1 above)
Comment (AAkira) [reply gradle/publish.gradle:94]: This bug is caused by bintray plugin.

 https://github.com/bintray/gradle-bintray-plugin/issues/229

**Comment by AAkira** [comment]
@terrakok 
Thank you for the pull request.
This PR is going to be closed because we need this using the latest Kotlin version and I want to change something.

**Pull request closed without merging**